### PR TITLE
Add react-testing-library example

### PR DESCRIPTION
**Pull request by jeanbauer**
This example show how you can test Next.js apps with [react-testing-library](https://github.com/kentcdodds/react-testing-library).

This library encourages your applications to be more accessible and allows you to get your tests closer to using your components the way a user will, which allows your tests to give you more confidence that your application will work when a real user uses it. And also, is a replacement for enzyme.

Image showing the words next.js + react testing library

### CODE REVIEW

File: examples/with-jest-react-testing-library/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "with-jest-react-testing-library",
+  "version": "1.0.0",
+  "dependencies": {
+    "next": "^7.0.2",
```
Comment (timneutkens) [inline examples/with-jest-react-testing-library/package.json]: Let's use `latest` like the other examples:

```suggestion
     "next": "latest",
```

File: examples/with-jest-react-testing-library/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "react": "16.7.0-alpha.2",
```
Comment (timneutkens) [inline examples/with-jest-react-testing-library/package.json]: Let's use stable React for the example, it doesn't seem like there is a strong requirement for React alpha:

```suggestion
     "react": "latest",
```

File: examples/with-jest-react-testing-library/package.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    "react-dom": "16.7.0-alpha.2"
```
Comment (timneutkens) [inline examples/with-jest-react-testing-library/package.json]:
```suggestion
     "react-dom": "latest"
```

File: examples/with-jest-react-testing-library/package.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  },
+  "devDependencies": {
+    "@babel/core": "^7.1.2",
```
Comment (timneutkens) [inline examples/with-jest-react-testing-library/package.json]: Is this needed? You'd override the version that is installed with Next.js

**Review by timneutkens — CHANGES_REQUESTED** [review]
Looks amazing, great example! Just had a few comments about the deps 💯

File: examples/with-jest-react-testing-library/package.json (hunk 4 above)
Comment (jeanbauer) [reply examples/with-jest-react-testing-library/package.json]: It is not needed, thanks for pointing out. :)

Comment (timneutkens) [reply examples/with-jest-react-testing-library/package.json]: 💯

**Review by timneutkens — APPROVED** [review]
Amazing 😌💯

**Pull request merged**